coverage.sh

```diff
@@ -1,19 +1,63 @@
-#!/bin/sh -ex
+#!/bin/bash
+#
+# Merges coverage reports from cypress and jest tests
+# and generates an HTML report to view and json report to submit to codecov
+#
+# For jest coverage to be collected and submitted correctly
+#  * Ensure the *coverageDirectory* is set to "coverage-jest" in packages.json
+#  * Ensure there is a npm "test" script
+#
+# For cypress collection to work correctly
+#  * Ensure "babel" (and dependencies) are at the latest (possible) version (aligned with frontend-components preferably)
+#  * Ensure the babel "istanbul" plugin is included and configured in babel.config.js
+#  * Ensure "nyc" is configuerd to output to "coverage-cypress"
+#  * Ensure there is a npm "test:ct" script
+#
+case $(uname) in
+Darwin)
+	export CODECOV_BIN="https://uploader.codecov.io/latest/macos/codecov"
+	;;
+Linux)
+	export CODECOV_BIN="https://uploader.codecov.io/latest/linux/codecov"
+	;;
+esac
 
-# Merge coverage reports
-# Currently Cypress coverage doesn't merge with anything
-# Keeping this for future expansion
-mkdir -p reports
-cp cypress-coverage/coverage-final.json reports/from-cypress.json
+rm -rf ./nyc_output
+rm -rf ./coverage-jest
+rm -rf ./coverage-cypress
+rm -rf ./coverage
 
-npx nyc merge reports .nyc_output/out.json
-npx nyc report --reporter json --report-dir coverage
+mkdir -p coverage/src
 
-# Upload the final coverage report to codecov https://docs.codecov.com/docs/codecov-uploader
-curl -Os https://uploader.codecov.io/latest/linux/codecov
-chmod +x codecov
+npm run test
+cp ./coverage-jest/coverage-final.json ./coverage/src/jest.json
 
-# Workaround for https://github.com/codecov/uploader/issues/475
-unset NODE_OPTIONS
+if [ -f cypress.config.js ]; then
+	npm run test:ct
+	cp ./coverage-cypress/coverage-final.json ./coverage/src/cypress.json
+else
+	echo "No cypress config found!"
+fi
 
-./codecov -t ${CODECOV_TOKEN} -f "coverage/coverage-final.json"
\ No newline at end of file
+npx nyc merge ./coverage/src ./coverage/coverage-final.json
+npx nyc report -t ./coverage --reporter html --report-dir ./coverage/html
+
+if [ -z "${TRAVIS}" ]; then
+	echo "Not on travis..."
+else
+	# Workaround for https://github.com/codecov/uploader/issues/475
+	unset NODE_OPTIONS
+	echo "Downloading codecov binary from $CODECOV_BIN"
+
+	curl -Os "$CODECOV_BIN"
+	chmod +x codecov
+
+	echo "Uploading to codecov"
+
+	./codecov --verbose -F "combined" -t "$CODECOV_TOKEN" -f ./coverage/coverage-final.json
+	./codecov --verbose -F "jest" -t "$CODECOV_TOKEN" -f ./coverage-jest/coverage-final.json
```

Comment:
We'll need to update the script in the interact-tools[0] (once we curl the script) to not try and upload jest results in apps that do not have jest results like this one. 

[0] https://github.com/RedHatInsights/insights-interact-tools/blob/main/scripts/coverage.sh